### PR TITLE
Add new config for Vxlan Vtep

### PR DESCRIPTION
**Pull request by saichint**
This PR is for adding new configurations for vxlan vtep and vxlan vtep vni.
Tests are run on relevant platforms and they are ok.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -5043,6 +5043,10 @@ Creates a VXLAN Network Virtualization Endpoint (NVE) overlay interface that ter
 |---------------------------------|--------------------------------------|
 | source_interface_hold_down_time | Not supported on N3k, N5k, N6k <br> Supported in OS Version 8.1.1 and later on N7k |
 | multisite_border_gateway_interface | Only supported on N9K-EX and N9K-FX devices. For eg: N9K-C93180YC-EX. Minimum OS version 7.0(3)I7(1) and minimum Module Version 1.9.0 |
+| global_suppress_arp             | Only supported on N9K and N9K-F running OS Version 9.2 and later |
```
Comment (mikewiebe) [inline README.md]: Here and elsewhere.  This should be `9.1 and later` right?

Also, I would not mention `N9k-F` in the documentation for any of these new properties.  There is really no distinction as far as this is concerned.  They are all N9k's running 9.1 or later.

File: README.md [hunk 2]
```diff
@@ -5052,6 +5056,18 @@ Determines whether or not the config should be present on the device. Valid valu
 ##### `description`
 Description of the NVE interface.  Valid values are string, or keyword 'default'.
 
+##### `global_ingress_replication_bgp`
+Configures ingress replication protocol to bgp for all VNIs. Valid values are true, false or keyword 'default'.
```
Comment (mikewiebe) [inline README.md]: Instead of `Configures` I would use `Sets ingress...`

File: lib/puppet/type/cisco_vxlan_vtep.rb [hunk 3]
```diff
@@ -130,10 +134,45 @@ def self.title_patterns
     end
   end # source_interface_hold_down_time
 
+  newproperty(:global_ingress_replication_bgp) do
+    desc "Configure ingress replication protocol to bgp. Valid values are true,
```
Comment (mikewiebe) [inline lib/puppet/type/cisco_vxlan_vtep.rb]: *nit /Configure/Sets/

File: tests/beaker_tests/cisco_vxlan_vtep_vni/test_vxlan_vtep_vni.rb [hunk 4]
```diff
@@ -293,6 +315,7 @@ def dependency_manifest(_tests, _id)
 def version_unsupported_properties(_tests, _id)
   unprops = {}
   unprops[:suppress_uuc] = '8.1.1' if platform[/n7k/]
+  unprops[:suppress_arp_disable] = '9.2' if platform[/n9k/]
```
Comment (mikewiebe) [inline tests/beaker_tests/cisco_vxlan_vtep_vni/test_vxlan_vtep_vni.rb]: `9.1`

File: README.md (hunk 1 above)
Comment (saichint) [reply README.md]: As mentioned in the NU, the CCO images are only 9.2.1 and 9.2.2, no 9.1

File: tests/beaker_tests/cisco_vxlan_vtep_vni/test_vxlan_vtep_vni.rb (hunk 4 above)
Comment (saichint) [reply tests/beaker_tests/cisco_vxlan_vtep_vni/test_vxlan_vtep_vni.rb]: See above

File: README.md (hunk 1 above)
Comment (saichint) [reply README.md]: About N9k vs N9k-F, there MUSt be a distinction some properties are supported on N9k and some on both N9k and N9k-F. If I remove it, then how do we say that some are not supported on N9k-F? We can discuss this if you are not clear

File: README.md (hunk 2 above)
Comment (saichint) [reply README.md]: Done

File: lib/puppet/type/cisco_vxlan_vtep.rb (hunk 3 above)
Comment (saichint) [reply lib/puppet/type/cisco_vxlan_vtep.rb]: Done

**Pull request merged**